pkg/operator/baremetal_pod.go

```diff
@@ -46,14 +56,56 @@ func setMariadbPassword() corev1.EnvVar {
 		ValueFrom: &corev1.EnvVarSource{
 			SecretKeyRef: &corev1.SecretKeySelector{
 				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "mariadb-password",
+					Name: baremetalSecretName,
 				},
-				Key: "password",
+				Key: baremetalSecretKey,
 			},
 		},
 	}
 }
 
+func generateRandomPassword() string {
+	rand.Seed(time.Now().UnixNano())
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
+		"abcdefghijklmnopqrstuvwxyz" +
+		"0123456789")
+	length := 16
+	buf := make([]rune, length)
+	for i := range buf {
+		buf[i] = chars[rand.Intn(len(chars))]
+	}
+	return (string(buf))
+}
+
+func createMariadbPasswordSecret(client coreclientv1.SecretsGetter, config *OperatorConfig) error {
+	glog.V(3).Info("Checking if the Maridb password secret already exists")
+	_, err := client.Secrets(config.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
```

Comment:
this could be inlined:
```
	if _, err := client.Secrets(config.TargetNamespace).Get(baremetalSecretName, metav1.GetOptions{}); err != nil {
		if apierrors.IsNotFound(err) {
			// Secret does not already exist. So, create one.
			mariadbPassword := generateRandomPassword()
			secret := &corev1.Secret{
				ObjectMeta: metav1.ObjectMeta{
					Name:      baremetalSecretName,
					Namespace: config.TargetNamespace,
				},
				StringData: map[string]string{
					baremetalSecretKey: mariadbPassword,
				},
			}

			glog.V(1).Infof("Maridb password does not exist so creating one.")
			_, err := client.Secrets(config.TargetNamespace).Create(secret)
			return err
		}
		glog.Errorf("Error while looking for existing Mariadb password : %v", err)
		return err
	}
```

Reply:
Done, I think.